packages/core-data/src/types/README.md

```diff
@@ -0,0 +1,211 @@
+
+For example, requesting `/wp/v2/posts/1?context=view` yields:
+
+```js
+{
+  "content": {
+    "protected": false,
+    "rendered": "\n<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n"
+  },
+  "title": {
+    "rendered": "Hello world!"
+  }
+  // other fields
+}
+```
+
+While requesting `/wp/v2/posts/1?context=edit`, yields:
+
+```js
+{
+  "content": {
+    "block_version": 1,
+    "protected": false,
+    "raw": "<!-- wp:paragraph -->\n<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n<!-- /wp:paragraph -->",
+    "rendered": "\n<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n"
+  },
+  "title": {
+    "raw": "Hello world!",
+    "rendered": "Hello world!"
+  }
+  // other fields
+}
+```
+
+And, finally, requesting `/wp/v2/posts/1?context=embed` yields:
+
+```js
+{
+  // Note content is missing
+  "title": {
+    "rendered": "Hello world!"
+  }
+  // other fields
+}
+```
+
+These contexts are supported by the core-data resolvers like `getEntityRecord()` and `getEntityRecords()` to retrieve the appropriate "flavor" of the data.
+
+The types describing different entity records must thus be aware of the relevant API context. This is implemented using the `Context` type parameter. For example, the implementation of the `Post` type resembles the following snippet:
+
+```ts
+import { OmitNevers } from "./helpers";
+
+interface BasePost<C extends Context> {
+	/**
+	 * A named status for the post.
+	 */
+	status: ContextualField< PostStatus, 'view' | 'edit', C >;
+
+	// ... other fields ...
+}
+
+export type Post<C extends Context> = OmitNevers<BasePost<C>>;
+```
+
+The `status` field of the `BasePost` is either a `PostStatus` when the Context is `view` or `edit`, or it's `never` when the Context is `embed`.
+
+The derived type `Post` is stripped of all the `never` fields, and so `Post<'embed'>` does not have the `status` type at all.
+
+### Static type checks for *edited* entity records, where certain fields become strings instead of objects.
+
+When the `post` is retrieved using `getEntityRecord`, its `content` field is an object:
+
+```js
+const post = wp.data.select('core').getEntityRecord( 'postType', 'post', 1, { context: 'view' } )
+// `post.content` is an object with two fields: protected and rendered
+```
+
+The block markup stored in `content` can only be rendered on the server so the REST API exposes both the raw markup and the rendered version. For example, `content.rendered` could used as a visual preview, and `content.raw` could be used to populate the code editor.
+
+The `content` field may still be updated via Javascript, but the updated content cannot be properly rendered without requesting the server. Therefore, it doesn't make sense to keep track the `rendered` when editing. The only remaining part is `raw`, and since that's a string, the entire edited `content` can be expressed a string.
```

Comment:
JavaScript

Reply:
this is a bit wordy or garden-pathy, distracting for what I think someone would care most about when reading it. like all writing, it may take a few different ideas to find a really good fit.

> When updating that field, however, all we can set is the raw value that the server will eventually render, so to do that the API expects us to send a much simpler `string` form which is the raw form that needs to be stored in the database. The types reflect this through the `Updatable<EntityRecord>` wrapper.

My formulation probably has glaring problems too, just trying to doodle on some different phrasings.

Reply:
it sounds quite good, thank you! I added a few tiny edits and a code snippet: https://github.com/WordPress/gutenberg/pull/38666/commits/3fe22976a4cd322ad1e5e51051de8e3976c7a94d